astropy/modeling/core.py

```diff
@@ -1068,6 +1080,49 @@ def has_user_bounding_box(self):
 
     # *** Public methods ***
 
+    def without_units_for_data(self, x, y, z=None):
+        """
+        Return an instance of the model for which the parameters have been
+        converted to the right units for the data, then the units have been
+        stripped away.
+        """
+
+        model = self.copy()
+
+        xunit = x.unit if isinstance(x, Quantity) else dimensionless_unscaled
+        yunit = y.unit if isinstance(y, Quantity) else dimensionless_unscaled
+        zunit = z.unit if z is not None and isinstance(z, Quantity) else dimensionless_unscaled
+
+        parameter_units = self._parameter_units_for_data_units(xunit, yunit, zunit)
+
+        for name, unit in parameter_units.items():
+            parameter = getattr(model, name)
+            if parameter.unit is not None:
+                parameter.value *= parameter.unit.to(unit)
```

Comment:
Not all transformations are multiplicative... Instead: `parameter.value = parameter.to(unit).value`.